pilot/pkg/networking/core/v1alpha3/httproute.go

```diff
@@ -463,7 +463,13 @@ func getVirtualHostsForSniffedServicePort(vhosts []*route.VirtualHost, routeName
 	}
 
 	if len(virtualHosts) == 0 {
-		virtualHosts = vhosts
+		return virtualHosts
+	} else if len(virtualHosts) == 1 {
+		virtualHosts[0].Domains = []string{"*"}
+	} else {
+		if features.EnableUnsafeAssertions {
+			panic(fmt.Sprintf("unexpectedly matched multiple virtual hosts for %v: %v", routeName, virtualHosts))
```

Comment:
nit: from the above this could never happen

Reply:
Why not? The conditions are `0`, `1`, and `>1`. It _shouldn't_ happen but due to a bug it could, which is why there is the assertion